### ISSUE TITLE
immich-machine-learning openvino not working (transcoding works)

### ISSUE THREAD

**Issue by Str1atum** [immich-app/immich#9523]
### The bug

Hi all,

I configured ubuntu in docker on Ubuntu 24.04 on a proxmox VM with PCIe passthrough for my Intel 13th gen Iris Xe
So far intel_gpu_top and hardware transcoding is working fine in immich (can see the utilization in intel_gpu_top and speed is fine) but I cannot get the ml part to use the GPU with openvino
The error in the logs is just:
WARNING  No GPU device found in OpenVINO. Falling back to CPU

### The OS that Immich Server is running on

Ubuntu 24.04

### Version of Immich Server

v1.105.1

### Version of Immich Mobile App

v1.105.1

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
immich-machine-learning:
    container_name: immich_machine_learning
    # For hardware acceleration, add one of -[armnn, cuda, openvino] to the image tag.
    # Example tag: ${IMMICH_VERSION:-release}-cuda
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}-openvino
    extends: # uncomment this section for hardware acceleration - see https://immich.app/docs/features/ml-hardware-acceleration
      file: hwaccel.ml.yml
      service: openvino # set to one of [armnn, cuda, openvino, openvino-wsl] for accelerated inference - use the `-wsl` version for WSL2 where applicable
    volumes:
      - model-cache:/cache
    env_file:
      - .env
    restart: unless-stopped
```

### Your .env content

```Shell
standard .env. just with folders / passwords changed
```

### Reproduction steps

```bash
see above
```

### Relevant log output

WARNING No GPU device found in OpenVINO. Falling back to CPU

**Comment by Str1atum** [comment]
It's i9-13900H

I tried with bookworm Debian 6.1 kernel but had no success at all.

**Comment by mertalev** [comment]
Hmm, you can try looking at some of the drivers mentioned in [this](https://docs.openvino.ai/2024/get-started/configurations/configurations-intel-gpu.html) page. [This](https://github.com/openvinotoolkit/docker_ci/blob/master/docs/accelerators.md) page also suggests using `--group-add`. For Compose files, this would mean first running `stat -c \"%g\" /dev/dri/render*`, then adding the output as:
```
...
group_add:
  - <output>

**Comment by Str1atum** [comment]
no success, driver installation went smooth but no changes.

I added this to my hwaccel.ml.ml

`openvino:
    device_cgroup_rules:
      - "c 189:* rmw"
    devices:
      - /dev/dri:/dev/dri
    group_add:
      - "993"
    volumes:
      - /dev/bus/usb:/dev/bus/usb`

**Comment by mertalev** [comment]
What's the output of `ls -l /dev/dri`? Also, can you set `LOG_LEVEL=debug`, try again and find the line `Available OpenVINO devices:` in the ML logs?

**Comment by Str1atum** [comment]
ls -l /dev/dri:

`total 0
drwxr-xr-x 2 root root        100 May 16 14:49 by-path
crw-rw---- 1 root video  226,   0 May 16 14:49 card0
crw-rw---- 1 root video  226,   1 May 16 14:49 card1
crw-rw---- 1 root render 226, 128 May 16 14:49 renderD128`

`[05/16/24 16:37:28] DEBUG    Available ORT providers:                           
                             {'OpenVINOExecutionProvider',                      
                             'CPUExecutionProvider'}                            
[05/16/24 16:37:28] DEBUG    Available OpenVINO devices: ['CPU']                
[05/16/24 16:37:28] WARNING  No GPU device found in OpenVINO. Falling back to   
                             CPU.                                               
[05/16/24 16:37:28] INFO     Setting 'nllb-clip-large-siglip__v1' execution     
                             providers to ['CPUExecutionProvider'], in          
                             descending order of preference                     
[05/16/24 16:37:28] DEBUG    Setting execution provider options to              
                             [{'arena_extend_strategy': 'kSameAsRequested'}]    
[05/16/24 16:37:28] DEBUG    Setting execution_mode to ORT_SEQUENTIAL           
[05/16/24 16:37:28] DEBUG    Setting inter_op_num_threads to 1                  
[05/16/24 16:37:28] DEBUG    Setting intra_op_num_threads to 2                  
[05/16/24 16:37:28] DEBUG    Setting preferred runtime to onnx                  
[05/16/24 16:37:28] INFO     Loading clip model 'nllb-clip-large-siglip__v1' to 
                             memory                                             `

**Comment by mertalev** [comment]
Can you also add the output of `stat -c \"%g\" /dev/dri/card*` to `group_add`? I think the OpenVINO container is non-root, so it may not be able to access the devices in `/dev/dri`.

**Comment by Str1atum** [comment]
![Screenshot 16 05 2024 um 21 50 06 PM](https://github.com/immich-app/immich/assets/76776216/dac5784d-ea18-429a-a0a3-d6048b4207d8)

![Screenshot 16 05 2024 um 21 50 20 PM](https://github.com/immich-app/immich/assets/76776216/af51e473-d7cf-4607-81f2-5ff61fce1686)

but still the same error

![Screenshot 16 05 2024 um 21 50 57 PM](https://github.com/immich-app/immich/assets/76776216/a6c20f3c-2959-4fd2-9e33-20515b13f2e1)

**Comment by Str1atum** [comment]
that seems to have worked

![Screenshot 16 05 2024 um 23 55 22 PM](https://github.com/immich-app/immich/assets/76776216/bbea6428-921d-4fc1-874f-05e3331d55ad)

**Comment by mertalev** [comment]
Sweet! That means it's a [bug](https://github.com/intel/compute-runtime/issues/710) in `intel-compute-runtime`.

**Comment by Str1atum** [comment]
So anything we can do about that or do I just need to keep that compose extension forever?

**Comment by mertalev** [comment]
We would need the [latest patched release](https://github.com/intel/compute-runtime/releases/tag/24.13.29138.7) for `intel-compute-runtime` in the image, which will take some time to be available in package managers. You can just keep those envs around until then.

**Comment by mertalev** [comment]
I decided to add these to the `hwaccel.ml.yml` file in the meantime: #9541.

**Comment by vuongtt92** [comment]
I added this to the openvino section in ` hwaccel.ml.yml` but it still doesn't work. Mine is intel N5015 installing Xpenology 7.2. Could you please help? The log said "no gpu device found in openvino"
    ```
environment:
      - NEOReadDebugKeys=1
      - OverrideGpuAddressSpace=48
```

**Comment by mertalev** [comment]
That fix only applies to kernels 6.7.5 or newer. I don't imagine a Synology server would have such a new kernel. It could be that the kernel is too old instead and/or that you're missing a driver.

**Comment by vuongtt92** [comment]
so there's nothing I can do about it? I see that hardware accelerator works well with Plex while transcoding but I've tried many fixes but can't get it work with immich ML
